react/components/SliderTrack.tsx

```diff
@@ -1,11 +1,41 @@
-import React, { FC } from 'react'
+import React, { FC, useEffect, useRef } from 'react'
 import { useListContext } from 'vtex.list-context'
 import { useCssHandles } from 'vtex.css-handles'
 
 import { useSliderState } from './SliderContext'
 
 const CSS_HANDLES = ['sliderTrack', 'slide', 'slideChildrenContainer']
 
+const isSlideVisible = (
+  index: number,
+  currentSlide: number,
+  slidesToShow: number
+): boolean => {
+  return index >= currentSlide && index < currentSlide + slidesToShow
+}
+
+const useSliderVisibility = (currentSlide: number, slidesPerPage: number) => {
+  /** Keeps track of slides that have been visualised before, to keep
+   * rendering them. We want to keep rendering them because the issue
+   * is mostly rendering slides that might never be viewed; On the other
+   * hand, hiding slides that were visible causes visual glitches */
```

Comment:
```suggestion
  /** Keeps track of slides that have been visualised before.
   * We want to keep rendering them because the issue
   * is mostly rendering slides that might never be viewed; On the other
   * hand, hiding slides that were visible causes visual glitches */
```

Reply:
💅